game/src/edit/roads.rs

```diff
@@ -421,21 +426,36 @@ fn make_main_panel(
         let stack_batch = stack.batch();
         let stack_bounds = stack_batch.get_bounds();
 
+        let mut rounding = CornerRadii::zero();
+        if idx == 0 {
+            rounding.top_left = DEFAULT_CORNER_RADIUS;
+            rounding.bottom_left = DEFAULT_CORNER_RADIUS;
+        }
+        if idx == lanes_len - 1 {
+            rounding.top_right = DEFAULT_CORNER_RADIUS;
+            rounding.bottom_right = DEFAULT_CORNER_RADIUS;
+        }
+
         current_lanes_ltr.push(
             ctx.style()
                 .btn_plain
                 .btn()
                 .image_batch(stack_batch, stack_bounds)
                 .disabled(Some(id) == current_lane)
                 .bg_color(ctx.style().section_bg, ControlState::Default)
+                .bg_color(ctx.style().section_bg.shade(0.1), ControlState::Hovered)
                 .bg_color(ctx.style().btn_solid_primary.bg, ControlState::Disabled)
+                .image_color(ctx.style().btn_plain.fg, ControlState::Disabled)
```

Comment:
The icon was faded for the currently selected one, so we have to also set the foreground properties for the disabled state.

One way to avoid these kinds of bugs might be to introduce a ControlState::Selected, but I'm not sure if the ramifications elsewhere would be worth it.

Reply:
There are only a handful of places abusing disabled to mean selected. I agree that it's probably not worth it yet